bootstrap/bootstrapper/fs/source.go

```diff
@@ -88,18 +71,25 @@ func (fss *fileSystemSource) GetAvailability(shard uint32, targetRangesForShard
 
 // ReadData returns raw series for a given shard within certain time ranges.
 func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.ShardResult, xtime.Ranges) {
-	log := fss.opts.GetLogger()
-
 	if xtime.IsEmpty(tr) {
 		return nil, tr
 	}
-	files, err := fss.getInfoFiles(shard)
-	if err != nil {
-		log.Errorf("unable to get info files for shard %d: %v", shard, err)
+
+	var files []string
+	fs.ForEachInfoFile(
+		fss.filePathPrefix,
+		shard,
+		func(fname string, _ []byte) {
+			files = append(files, fname)
+		},
+	)
+	if len(files) == 0 {
 		return nil, tr
 	}
+
+	log := fss.opts.GetLogger()
```

Comment:
Perhaps at start of the method so if log statements added later near top someone doesn't have to move this statement?